plugin/plugin.go

```diff
@@ -0,0 +1,160 @@
+package plugin
+
+import (
+	"errors"
+)
+
+// NamePrefix is the prefix required on all plugin binary names.
+const NamePrefix = "notation-"
+
+// Command is a CLI command available in the plugin contract.
+type Command string
+
+const (
+	// CommandGetMetadata is the name of the plugin command
+	// which must be supported by every plugin and returns the
+	// plugin metadata.
+	CommandGetMetadata Command = "get-plugin-metadata"
+
+	// CommandGenerateSignature is the name of the plugin command
+	// which must be supported by every plugin that has the
+	// SIGNATURE_GENERATOR capability.
+	CommandGenerateSignature Command = "generate-signature"
+
+	// CommandGenerateEnvelope is the name of the plugin command
+	// which must be supported by every plugin that has the
+	// SIGNATURE_ENVELOPE_GENERATOR capability.
+	CommandGenerateEnvelope Command = "generate-envelope"
+)
+
+// Capability returns the capability associated to the command.
+func (c Command) Capability() Capability {
+	switch c {
+	case CommandGenerateSignature:
+		return CapabilitySignatureGenerator
+	case CommandGenerateEnvelope:
+		return CapabilityEnvelopeGenerator
+	default:
+		return ""
+	}
+}
+
+// Capability returns the response associated to the command.
+func (c Command) NewResponse() interface{} {
+	switch c {
+	case CommandGetMetadata:
+		return new(Metadata)
+	case CommandGenerateSignature:
+		return new(GenerateSignatureResponse)
+	case CommandGenerateEnvelope:
+		return new(GenerateEnvelopeResponse)
+	default:
+		return nil
+	}
+}
+
+// Capability is a feature available in the plugin contract.
+type Capability string
+
+const (
+	// CapabilitySignatureGenerator is the name of the capability
+	// which should support a plugin to support generating signatures.
+	CapabilitySignatureGenerator Capability = "SIGNATURE_GENERATOR"
+
+	// CapabilityEnvelopeGenerator is the name of the capability
+	// which should support a plugin to support generating envelope signatures.
+	CapabilityEnvelopeGenerator Capability = "SIGNATURE_ENVELOPE_GENERATOR"
+)
+
+// GenerateSignatureRequest contains the parameters passed in a generate-signature request.
+// All parameters are required.
+type GenerateSignatureRequest struct {
+	ContractVersion string
+	KeyName         string
+	KeyID           string
+}
```

Comment:
Is missing the `payload` field from [spec](https://github.com/notaryproject/notaryproject/blob/main/specs/plugin-extensibility.md#user-content-signature-generator). I'm assuming that KeyName and KeyID are in place of KeyDefinition.